src/framework/COScript.m

```diff
@@ -405,6 +405,56 @@ - (void)deleteObjectWithName:(NSString*)name {
     [_mochaRuntime removeObjectWithName:name];
 }
 
+# pragma mark - require
+
+- (NSURL*)resolveModule:(NSString *)module currentURL:(NSURL*)currentURL {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    BOOL isRelative = [module characterAtIndex:0] == '.';
+    NSString* modulePath = [module stringByStandardizingPath];
+    NSURL* moduleURL = isRelative ? [NSURL URLWithString:modulePath relativeToURL:currentURL] : [NSURL fileURLWithPath:modulePath];
+    BOOL isDir;
+    
+    if ([fileManager fileExistsAtPath:moduleURL.path isDirectory:&isDir]) {
```

Comment:
Rather than querying if a file exists and is a directory which is a bit error prone, it's often better to do something more specific such as:

`[url checkResourceIsReachableAndReturnError:NULL]`
`[url getResourceValue:&isDirectory forKey:NSURLIsDirectoryKey error:NULL]`

Perhaps you want to do so here? For example, the latter will accurately tell you if the path exists and is a directory.

Reply:
I'm confused, I do want to check if a file exists and is a directory. Why is it error prone?

Reply:
Ah got it. Yeah I think it's fine, it will blow up like it should if it can't be accessed